drivers/health_flexible_service/app/models/health_flexible_service/member_list.rb

```diff
@@ -0,0 +1,87 @@
+###
+# Copyright 2016 - 2020 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/master/LICENSE.md
+###
+
+module HealthFlexibleService
+  class MemberList
+    include ArelHelper
+
+    attr_accessor :r_number, :end_date
+
+    def initialize(aco_id, r_number, end_date)
+      @aco_id = aco_id
+      @r_number = r_number
+      @report_range = ((end_date - 3.months)..end_date)
+    end
+
+    def filename
+      @filename ||= begin
+        aco = Health::AccountableCareOrganization.find(@aco_id)
+        "#{aco.short_name}_ML_R#{@r_number}_QE#{@report_range.end.strftime('%Y%m%d')}.xlsx".upcase
+      end
+    end
+
+    def write_to(workbook)
+      categories.each do |worksheet_name, category_name|
+        worksheet = workbook[worksheet_name]
+        vprs = vpr_scope(category_name)
+        vprs.each_with_index do |vpr, row_index|
+          columns(vpr, category_name).each_with_index do |value, col_index|
+            worksheet.add_cell(row_index + 1, col_index, value)
+          end
+        end
+      end
+    end
+
+    def categories
+      @categories ||= {
+        'PRETENANCY_INDIVIDUAL' => 'Pre-Tenancy Supports: Individual Supports',
+        'PRETENANCY_TRANSITIONAL' => 'Pre-Tenancy Supports: Transitional Assistance',
+        'TENANCY_SUSTAINING' => 'Tenancy Sustaining Supports',
+        'HOME_MODIFICATIONS' => 'Home Modification',
+        'NUTRITION' => 'Nutritional Sustaining Supports',
+      }.freeze
+    end
+
+    def columns(vpr, category)
+      [
+        vpr.patient[:medicaid_id],
+        vpr[:last_name],
+        vpr[:first_name],
+        vpr[:middle_name],
+        '', # We don't collect suffix, so leave it blank
+        vpr[:dob].strftime('%Y%m%d'),
+        vpr.patient.aco[:vpr_name],
+        delivery_entities(vpr, category),
+        'No', # transportation
+        vpr[:gender],
+        vpr[:sexual_orientation],
+        vpr[:race],
+        vpr[:primary_language],
+        vpr[:education],
+        vpr[:employment],
+      ]
+    end
+
+    def vpr_scope(category)
+      HealthFlexibleService::Vpr.
+        joins(patient: :patient_referral).
+        preload(:patient, patient: :patient_referral).
```

Comment:
You should only need `preload(patient: :patient_referral)` the first `:patient` is redundant.